crates/api/src/local_user/update_totp.rs

```diff
@@ -0,0 +1,61 @@
+use crate::check_totp_2fa_valid;
+use actix_web::web::{Data, Json};
+use lemmy_api_common::{
+  context::LemmyContext,
+  person::{UpdateTotp, UpdateTotpResponse},
+};
+use lemmy_db_schema::{
+  source::local_user::{LocalUser, LocalUserUpdateForm},
+  traits::Crud,
+};
+use lemmy_db_views::structs::{LocalUserView, SiteView};
+use lemmy_utils::error::{LemmyError, LemmyErrorType};
+
+/// Enable or disable two-factor-authentication. The current setting is determined from
+/// [LocalUser.totp_2fa_enabled].
+///
+/// To enable, you need to first call [generate_totp_secret] and then pass a valid token to this
+/// function.
+///
+/// Disabling is only possible if 2FA was previously enabled. Again it is necessary to pass a valid
+/// token.
+#[tracing::instrument(skip(context))]
+pub async fn update_totp(
+  data: Json<UpdateTotp>,
+  local_user_view: LocalUserView,
+  context: Data<LemmyContext>,
+) -> Result<Json<UpdateTotpResponse>, LemmyError> {
+  let site_view = SiteView::read_local(&mut context.pool()).await?;
+
+  // require valid 2fa token to enable or disable 2fa
+  if local_user_view.local_user.totp_2fa_secret.is_none() {
+    return Err(LemmyErrorType::MissingTotpToken.into());
+  }
+  check_totp_2fa_valid(
+    &local_user_view,
+    &Some(data.totp_token.clone()),
+    &site_view.site.name,
+  )?;
+
+  // toggle the 2fa setting
+  let mut local_user_form = LocalUserUpdateForm {
+    totp_2fa_enabled: Some(data.enabled),
+    ..Default::default()
+  };
+
+  // clear totp secret if 2fa is being disabled
+  if !data.enabled {
+    local_user_form.totp_2fa_secret = None;
```

Comment:
You can move this into the LocalUserUpdateForm, and not have to have that mut. And `Some(None)` clears it, `None` means *don't update*. 

So `totp_2fa_secret: enabled { None } else { Some(None)`